### ISSUE TITLE
gmsh not mentioned as dependency

### ISSUE THREAD

**Issue by MuellerSeb** [nschloe/pygmsh#402]
When installing `pygmsh` I can't execute any example:
```python
ModuleNotFoundError: No module named 'gmsh'
```
It should be added in `setup.cfg`.

**Comment by nschloe** [comment]
I'm a bit on the edge on this. The problem is that gmsh doesn't properly update their pip resources, and their pip package a bit weird anyway. (For example, they don't include all sources, but have a separate process fetch gmsh from their site during installation.) The recommended installation -- and it's always been that way -- is to use gmsh from the system.

**Comment by MuellerSeb** [comment]
So I am currently unable to use `pygmsh` on Ubuntu 18.04?

**Comment by gdmcbain** [comment]
Well, there is still [gmsh-sdk](https://pypi.org/project/gmsh-sdk/).  I know it says it's deprecated, but on 18.04.5 LTS, with Miniconda:
```shell
conda create -n pygmsh-402 pip
conda activate pygmsh-402
pip install gmsh-sdk pygmsh
gmsh -version # 4.6.0
```
and then 
```python
from pygmsh.geo import Geometry
with Geometry() as geom:
     geom.add_circle([0]*2, 1)
     mesh = geom.generate_mesh()
mesh
```

All works.  I agree with @nschloe that 
>  gmsh doesn't properly update their pip resources, and their pip package a bit weird 

Using gmsh from the system in the sense of `apt install gmsh` is no good on Ubuntu 18.04.5 LTS since that's Gmsh 3.0.6 which wouldn't even have any Python bindings, I don't think, and would be too old anyway.  I assume it's possible to install Gmsh outside Python and somehow set up the PYTHONPATH but it's not obvious and I don't know how to do it.

**Comment by nschloe** [comment]
You _can_ also install gmsh from pypi, it just isn't a hard dependency.

**Comment by MuellerSeb** [comment]
But following the `import` statements, it seems to be a hard dependency AFAICT.

**Comment by nschloe** [comment]
Yeah, you need gmsh. It's just that I wouldn't recommend installing from pypi. The installation method is that you get an "installer" package which checks your platform and immediately downloads pre-built sources from another server. I've never seen any package do that and it seems like another weird thing that only gmsh does. If you want that, you can `pip install gmsh`. If not, use your system package. I wouldn't want to pygmsh to make that decision for its users.

**Comment by J-N-ch** [comment]
[](https://github.com/J-N-ch/gmsh_docker_env)
This is how I managed to solve the problem with the help of the suggestions given by you guys.
Thanks a lot !

**Comment by VikingScientist** [comment]
I solved this by using the first workaround listed in this thread
https://gitlab.onelab.info/gmsh/gmsh/-/issues/1023

> 
> Right now, as a workaround, you can eiter :
> 
> * add `/usr/local/lib/python3.8/site-packages/gmsh-4.6.0-Linux64-sdk/lib` to your `PYTHONPATH`
> * or re-install gmsh with `--user` (gmsh executable script will be in $HOME/.local/bin):
> 
> ```
> sudo pip3 uninstall gmsh
> pip3 install --user gmsh
> ```

**Comment by nschloe** [comment]
@VikingScientist You don't need `--user` if you have a reasonably recent pip.

**Comment by VikingScientist** [comment]
For Ubuntu you do. On my windows setup then simply `pip install gmsh` installed gmsh correctly, but for my Ubuntu setup then I needed to manually fix this somehow. I edited `PYTHONPATH`, but from the thread linked above there is reported that installing with `--user` also solves it. I haven't personally tested the latter solution.

**Comment by nschloe** [comment]
> For Ubuntu you do.

If you do, it's because your Python install is misconfigured somehow. `--user` is default since 2019 (see https://github.com/pypa/pip/pull/7002).

**Comment by kayarre** [comment]
I must have missed a change, as I thought I could set an environment variable and use the downloaded binaries of gmsh?

**Comment by surendranlu** [comment]
I installed GMSH from source using the following commands in my WSL2
```
    $ cd gmsh-4.6.0-source
    $ mkdir build
    $ cd build
    $ cmake ..
    $ sudo make
    $ sudo make install
```
However when I tried ```gmsh --version``` I got the following error code ```/usr/bin/env: ‘python’: No such file or directory```
Then I tried ``sudo apt install python-is-python3``` then I got another error
```
Traceback (most recent call last):
  File "/mnt/c/Users/USERX/AppData/Roaming/Python/Python39/Scripts/gmsh", line 4, in <module>
    import gmsh
ModuleNotFoundError: No module named 'gmsh'
```
How to get around this?
